### DISCUSSION TITLE
Attaching on container start when using the API.

### DISCUSSION THREAD

**Discussion by peter-facko** [containers/podman#21254] [Q&A] [answered]
Hello,

is there any way to attach the standard IO when starting a container using the API?

I thought of starting the container as paused and then attaching with a separate request
but that doesn't seem to be possible.

The only alternative I see now is creating the container with a sleep process
and then execing the actual process. That depends on the container having some
sleep executable however.

In general the use case is the same as would be for the `podman container start` `--attach` option in the CLI.

Thank you in advance for any help.

**Comment by rhatdan** [comment, marked as answer]
yes you can use the attach API which is what `podman --remote container start --attach` is using.

In API testing I see:

20-containers.at:t POST "containers/foo/attach?logs=true&stream=false" 200 \
20-containers.at:t POST "/v1.42/containers/foo/attach?logs=true&stream=false" 200
20-containers.at:t POST "/v4.6.0/libpod/containers/foo/attach?logs=true&stream=false" 200
20-containers.at:t POST "/v4.7.0/libpod/containers/foo/attach?logs=true&stream=false" 200

I guess you might want to set stream=true.
